serve_test.go

```diff
@@ -188,6 +189,46 @@ func TestServer_Run(t *testing.T) {
 	})
 }
 
+func TestServer_RunTLS(t *testing.T) {
+	// This is not a standard test, it is here to ensure that the server can run.
+	// Please do not run this kind of test for your controllers, it is NOT unit testing.
+	tt := []struct {
+		name string
+		opts []func(*Server)
+	}{
+		{
+			name: "can run TLS server with TLS config",
+			opts: []func(*Server){WithTLSConfig(&tls.Config{}), WithoutLogger()},
+		},
+		{
+			name: "can run TLS server with TLS files",
```

Comment:
This case also never runs ListenAndServerTLS as you're checking for `!= ""` in `isTLS`

Reply:
Didn't see the forest from the trees. Sorry, I'll fix ths.

Reply:
I might have exaggerated with a full blown functional test for the TLS server, but since I messed up on my first attempt I wanted to stop making any assumptions and make sure everything actually works.
Looking forward to any feedback that would help simplifying the TLS server tests.